aten/src/ATen/native/cuda/layer_norm_kernel.cu

```diff
@@ -171,14 +196,21 @@ __device__ WelfordDataLN compute_stats(
       vec_t data = X_vec[i];
       #pragma unroll
       for (int ii=0; ii < vec_size; ii++){
-        wd = cuWelfordOnlineSum(static_cast<acc_t>(data.val[ii]), wd);
+        if constexpr (!rms_norm){
```

Comment:
nit: will look slightly better if you hide this conditional inside templated `cuOnlineSum` function (template on whether it's rms)